File: Python/array/findMaxConsecutiveOnes.py
class Solution:
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l = len(nums)
        max_cnt, cnt = 0, 0
        for i in range(l):
            if nums[i] == 1:
                cnt += 1
            else:
                cnt = 0
            if max_cnt < cnt:
                max_cnt = cnt
                
        return max_cnt
                

File: JavaScript/array/findMaxConsecutiveOnes.js
/**
 * 
 * 打擂台
 * 
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
  if (nums === null || nums.length === 0) {
    return 0;
  }
  let max = 0;
  let sum = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === 1) {
      sum++;
    } else {
      max = max > sum ? max : sum;
      sum = 0;
    }
  }
  return max > sum ? max : sum;
};